src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        tail = new Node<>(tail, value, null);
+
+        if (size == 0) {
+            head = tail;
+        } else {
+            tail.prev.next = tail;
+        }
+
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
not a mistake, but I'd rather add return after add and unwrap the remaining part of the code